### ISSUE TITLE
iOS a11y Text input

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#12786]
- [x] Text fields are not announced as "Text field"
- [x] The hint "double tap to edit" is missing
- [x] content of text field is not announced
- [x] It is not announced weather a text field is in edit mode
- [x] The insert mode (e.g. character mode) is not announced.
- [x] The position of the cursor is not announced 
- [x] Entered characters are not echoed back
- [x] Swiping up or down doesn't move the courser
- [x] Selecting text with reverse-pinch doesn't work
- [x] After inserting space the last word is not echoed back
- [ ] C&P text when rotor is in edit mode

**Comment by goderbauer** [comment]
Related: Android text input (https://github.com/flutter/flutter/issues/12785)

**Comment by goderbauer** [comment]
This one could be tricky. Looks like Apple doesn't offer an API to mark something as a TextField for accessibility purposes.

**Comment by eseidelGoogle** [comment]
Please be sure to file radars for this and any other missing APIs we run into: bugreport.apple.com

**Comment by yjbanov** [comment]
Since a lot of the items here are done, let's close this issue and focus on fixing the follow-up ones, like https://github.com/flutter/flutter/issues/14543, https://github.com/flutter/flutter/issues/15224, https://github.com/flutter/flutter/issues/15188 and [everything under accessibility that's "text"](https://github.com/flutter/flutter/issues?utf8=%E2%9C%93&q=is%3Aopen+label%3A%22prod%3A+accessibility%22+text).
